### PR TITLE
Orange.data: Clean up the namespace

### PR DESCRIPTION
**Pull request by lanzagar**
##### Issue

Orange.data namespace included a lot of junk.

##### Description of changes
Define `__all__` in data.io.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/data/io.py [hunk 1]
```diff
@@ -29,6 +29,9 @@
 from Orange.util import Registry, flatten, namegen
 
 
+__all__ = ["Flags"]
```
Comment (kernc) [inline Orange/data/io.py]: I don't think that's ok. The module defines a lot more "public" stuff than just `Flags`. Perhaps rather make `import *` in _Orange.data.\_\_init\_\_.py_ explicit?

Comment (astaric) [inline Orange/data/io.py]: FileFormat might be considered public as well, as someone might use it to define their own readers, but what else? Reader implementations are not supposed to be used directly, they are accessible through FileReader.get_reader.

Comment (lanzagar) [inline Orange/data/io.py]: Added `FileFormat`. As said, the individual readers are really just current implementations that should not be used directly elsewhere, so I don't think they belong in `__all__`. And they can be imported from data.io if needed (for exceptional cases only).

Comment (kernc) [inline Orange/data/io.py]: >  but what else?

@astaric always more than one thinks.

https://github.com/biolab/orange3-text/pull/176/files

**Pull request merged**